e3db/types/note_options.py

```diff
@@ -0,0 +1,81 @@
+from datetime import date
+
+# Look for a possible alternative to a NoteOptions class. Not a huge priority at the moment, 
```

Comment:
Exactly, as you’re building out methods such as write note, read note, login etc that have to interact with Notes, and have to work with multiple similar but subtly different note types (anonymous, premium, non-premium), you will gain the experience in the ergonomics of creating notes and I think we can break out this generic note options type into more focused sub types that are based more on how the sdk user interacts with notes and less on “here’s a grab bag of values some of which are crucial and some of which are only needed in certain situations that aren’t obvious to anyone that didn’t write this code”

Reply:
Even if there is nothing obvious or worthwhile to refine with this or other note types, the critical thinking on user focused software design is valuable in itself and a good software engineering muscle to train

Reply:
Makes sense! I can leave a mention of this in its documentation to be a reminder for us of the intention to change its design when we start to approach different note types (so, the near future). Until then I think I would be getting ahead of myself by spending time splitting it versus ensuring there is a note object for others working on the Python SDK to use more immediately.

Reply:
We're in peaceful agreement, we're both saying "keep this in mind but not try to shine the penny before it is minted"